tests/Unit/PointwiseFunctions/AnalyticSolutions/Test_Protocols.cpp

```diff
@@ -0,0 +1,145 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Protocols.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "tests/Unit/ProtocolTestHelpers.hpp"
+
+namespace {
+
+struct SomeField : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+struct AnotherField : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+
+}  // namespace
+
+namespace test_evolution_protocols_analytic_solution {
+
+/// [evolution_analytic_sol_example]
+template <size_t Dim>
+struct EvolutionAnalyticSolution : evolution::protocols::AnalyticSolution {
+ private:
+  using PrecomputedData = double;
+
+  // Functions computing individual variables
+  tuples::TaggedTuple<SomeField> variables(
+      const tnsr::I<DataVector, Dim>& x, const double t,
+      const PrecomputedData& precomputed_data,
+      tmpl::list<SomeField> /*meta*/) noexcept {
+    return {Scalar<DataVector>{precomputed_data * t * get<0>(x)}};
```

Comment:
This might not need definitions of the functions in order to pass the protocol checks (because they're all compile-time)

Reply:
Indeed it doesn't! Everything but the function declaration can be removed. What do you think, should the test provide the implementation nonetheless so it appears in the documentation and gives an example, or should it be reduced to only the function declaration?